compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -391,13 +391,16 @@ fn run_compiler(
 
         let linker = compiler.enter(|queries| {
             let early_exit = || early_exit().map(|_| None);
+
+            // Parse the crate root source code (doesn't parse submodules yet)
```

Comment:
As far as I can tell, this does the source of the entire crate.

Reply:
The comment is correct, though I would maybe say "crate root source file" instead of "crate root source code". Everything other than the root source file is only getting parsed during macro expansion.

Reply:
Oh, interesting! In that case that should be mentioned.

Reply:
> crate root source file

The reason I went with "source code" instead of file is because technically you don't need a file as you can parse a regular string.

Reply:
I've mentioned that everything else is getting done later. 
I wonder if the rustc-dev-guide says this?